### PR TITLE
fix: Optimizations

### PR DESCRIPTION
**Pull request by aminya**
Optimizes minimap-find-and-replace:
- Replaces Object with Map which results in faster loops, clearing, etc
- Adds CSS containment

This is ready for now. I added a big file for testing. Currently, adding/removing decorations is pretty fast. We can do more optimization later if needed.

### CODE REVIEW

File: lib/minimap-find-and-replace-binding.js [hunk 1]
```diff
@@ -49,16 +49,17 @@ class MinimapFindAndReplaceBinding {
 	}
 
 	clear () {
-		let id
-		for (id in this.subscriptionsByMarkerId) {
-			const sub = this.subscriptionsByMarkerId[id]
-			sub.dispose()
+		const subscriptionsByMarkerIdKeys = Object.keys(this.subscriptionsByMarkerId)
+		for (let i = 0, len = subscriptionsByMarkerIdKeys.length; i < len; i++) {
```
Comment (UziTech) [inline lib/minimap-find-and-replace-binding.js:53]: Is this for loop really faster than for in? I feel like V8 would be able to optimize a for in loop a lot easier.

Comment (aminya) [reply lib/minimap-find-and-replace-binding.js:53]: Yes, it is faster. 
https://github.com/aminya/typescript-optimization

Comment (aminya) [reply lib/minimap-find-and-replace-binding.js:55]: I will just replace it with Map. Object should not be used in these cases.

**Comment by aminya** [comment]
This is ready for now. I added a big file for testing. Currently, the

**Comment by aminya** [comment]
@UziTech Could you invite me to the organization? Although I have full access to minimap repo, I am not a member!

**Comment by UziTech** [comment]
I don't have access to the organization. I just have access to each repo. I think if we are both going to be updating these repos we need to make sure each PR has at least one approval from a person that did not create the PR.

**Comment by aminya** [comment]
OK. No problem. I was just confused about why I am not shown in the member list of the organization.

File: package.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
   },
   "devDependencies": {
     "@semantic-release/apm-config": "^8.0.0",
+    "@types/atom": "^1.40.5",
```
Comment (UziTech) [inline package.json:41]: this package isn't written in typescript is this necessary?

Comment (aminya) [reply package.json:41]: Datatips in JavaScript come from TypeScript. JavaScript itself is not smart enough to know what is going on in the code. 

Here, I was checking some of the API of Atom to see where decorate API comes from.

Comment (UziTech) [reply package.json:41]: Can we remove it? I don't see how it helps.

Comment (aminya) [reply package.json]: It helps the datatips. I reverted it anyway.

**Comment by UziTech** [comment]
If you want to we could convert these packages to typescript.

**Pull request merged**